Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs

```diff
@@ -0,0 +1,244 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System.Collections;
+using HoloToolkit.Unity;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// The SpatialUnderstanding class controls the state and flow of the 
+    /// scanning process used in the understanding module. 
+    /// </summary>
+    [RequireComponent(typeof(SpatialUnderstandingSourceMesh))]
+    [RequireComponent(typeof(SpatialUnderstandingCustomMesh))]
+    public class SpatialUnderstanding : Singleton<SpatialUnderstanding>
+    {
+        // Consts
+        public const float ScanSearchDistance = 8.0f;
+
+        // Enums
+        public enum ScanStates
+        {
+            None,
+            ReadyToScan,
+            Scanning,
+            Finishing,
+            Done
+        }
```

Comment:
C# [naming conventions](https://msdn.microsoft.com/en-us/library/ms229040(v=vs.110).aspx) dictate that this enum should be singular (ScanState) since an instance of it will never contain more than one state (Flags). 

> ✓ DO use a singular type name for an enumeration unless its values are bit fields.
> ✓ DO use a plural type name for an enumeration with bit fields as values, also called flags enum.